x/dkim/keeper/msg_server.go

```diff
@@ -0,0 +1,87 @@
+package keeper
+
+import (
+	"context"
+
+	"cosmossdk.io/errors"
+
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+
+	dkimv1 "github.com/burnt-labs/xion/api/xion/dkim/v1"
+	"github.com/burnt-labs/xion/x/dkim/types"
+)
+
+type msgServer struct {
+	k Keeper
+}
+
+var _ types.MsgServer = msgServer{}
+
+// NewMsgServerImpl returns an implementation of the module MsgServer interface.
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{k: keeper}
+}
+
+func (ms msgServer) UpdateParams(ctx context.Context, msg *types.MsgUpdateParams) (*types.MsgUpdateParamsResponse, error) {
+	if ms.k.authority != msg.Authority {
+		return nil, errors.Wrapf(govtypes.ErrInvalidSigner, "invalid authority; expected %s, got %s", ms.k.authority, msg.Authority)
+	}
+
+	return nil, ms.k.Params.Set(ctx, msg.Params)
+}
+
+func SaveDkimPubKey(ctx context.Context, dkimKey types.DkimPubKey, store dkimv1.StateStore) (bool, error) {
+	if err := store.DkimPubKeyTable().Save(ctx, &dkimv1.DkimPubKey{
+		Domain:       dkimKey.Domain,
+		PubKey:       dkimKey.PubKey,
+		Selector:     dkimKey.Selector,
+		PoseidonHash: dkimKey.PoseidonHash,
+		Version:      dkimv1.Version(dkimKey.Version),
+		KeyType:      dkimv1.KeyType(dkimKey.KeyType),
+	}); err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+func SaveDkimPubKeys(ctx context.Context, dkimKeys []types.DkimPubKey, store dkimv1.StateStore) (bool, error) {
+	for _, dkimKey := range dkimKeys {
+		if isSaved, err := SaveDkimPubKey(ctx, dkimKey, store); !isSaved {
+			return false, err
+		}
+	}
+	return true, nil
+}
+
+// AddDkimPubKey implements types.MsgServer.
+func (ms msgServer) AddDkimPubKey(ctx context.Context, msg *types.MsgAddDkimPubKeys) (*types.MsgAddDkimPubKeysResponse, error) {
```

Comment:
Does this add one key or multiple? The name is confusing

Reply:
It should be multiple